src/mono/mono/mini/simd-intrinsics.c

```diff
@@ -1820,27 +1817,22 @@ emit_sri_vector (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSignature *fsi
 		and_res_vec->sreg2 = msb_mask_vec->dreg;
 
 		MonoInst* msb_shift_vec = emit_msb_shift_vector_constant (cfg, arg_class, arg0_type);
-		MonoInst* shift_res_vec = emit_simd_ins (cfg, arg_class, OP_ARM64_USHL, and_res_vec->dreg, msb_shift_vec->dreg);
+		
+		MonoInst* shift_res_vec = emit_simd_ins (cfg, arg_class, OP_XOP_OVR_X_X_X, and_res_vec->dreg, msb_shift_vec->dreg);
+		shift_res_vec->inst_c0 = INTRINS_AARCH64_ADV_SIMD_USHL;
 		shift_res_vec->inst_c1 = arg0_type;
 
 		if (arg0_type == MONO_TYPE_I1 || arg0_type == MONO_TYPE_U1) {
-			// Always perform usigned operations as vector sum and extract operations could sign-extend the result into the GP register
+			// Always perform unsigned operations as vector sum and extract operations could sign-extend the result into the GP register
 			// making the final result invalid. This is not needed for wider type as the maximum sum of extracted MSB cannot be larger than 8bits
 			arg0_type = MONO_TYPE_U1;
 
-			// In order to sum high and low 64bits of the shifted vector separatly, we use a zeroed vector and the extract operation
-			MonoInst* zero_vec = emit_xzero(cfg, arg_class);
-
-			MonoInst* ext_low_vec = emit_simd_ins (cfg, arg_class, OP_ARM64_EXT_IMM, zero_vec->dreg, shift_res_vec->dreg);
-			ext_low_vec->inst_c0 = 8;
-			ext_low_vec->inst_c1 = arg0_type;
+			MonoInst* ext_low_vec = emit_simd_ins_for_sig (cfg, arg_class, OP_XLOWER, 8, arg0_type, fsig, &shift_res_vec);
 			MonoInst* sum_low_vec = emit_sum_vector (cfg, fsig->params [0], arg0_type, ext_low_vec);
-
-			MonoInst* ext_high_vec = emit_simd_ins (cfg, arg_class, OP_ARM64_EXT_IMM, shift_res_vec->dreg, zero_vec->dreg);
-			ext_high_vec->inst_c0 = 8;
-			ext_high_vec->inst_c1 = arg0_type;
-			MonoInst* sum_high_vec = emit_sum_vector (cfg, fsig->params [0], arg0_type, ext_high_vec);
-
+			
+			MonoInst* ext_high_vec = emit_simd_ins_for_sig (cfg, arg_class, OP_XUPPER, 8, arg0_type, fsig, &shift_res_vec);
+			MonoInst* sum_high_vec = emit_sum_vector (cfg, fsig->params [0], arg0_type, ext_high_vec);			
+			
```

Comment:
This now looks much cleaner and thanks for looking into ways of unifying the code for both backends! 
However, I would like to draw attention to:
> The change removed 1xext.16b instruction and added two new ones (mov.8b and eor.16b). If this change introduces any significant regressions we can revert the change for miniJIT.

as we now generate an extra instruction in mini-JIT case, we should run the microbenchmarks with/without this PR to see how much that affects the overall performance. After which we can decide whether to revert the mini-jit change or keep it (as you rightfully suggested).

Additionally, if I am reading the LLVM AOT output correctly, LLVM managed to understand that `OP_XLOWER` + `OP_XUPPER` can be covered with `ext.16b`, could we do the same in case of mini-jit?

Reply:
I verified the potential regression locally using microbenchmarks and there seems to be no regression with the new version. In the table below are times required for running 1000x `Vector128<Byte>.ExtractMostSignificantBits()`.

|                              Method |     Mean |     Error |    StdDev |   Median |      Min |      Max |
|------------------------------------ |---------:|----------:|----------:|---------:|---------:|---------:|
| new_miniJIT | **1.057 μs** | 0.0080 μs | 0.0075 μs | 1.058 μs | 1.039 μs | 1.068 μs |
| old_miniJIT | 1.090 μs | 0.0510 μs | 0.0587 μs | 1.117 μs | 0.9835 μs | 1.154 μs | 
| SW_implem. | 14.17 μs | 0.682 μs | 0.785 μs | 14.64 μs | 12.73 μs | 14.86 μs |

I'm not sure how much work would be required to optimize miniJIT to work similarly to LLVM as I don't orient myself well enough inside its code at the moment. We can keep it as an open issue for the future.